server/message_hooks.go

```diff
@@ -14,13 +14,28 @@ import (
 	"github.com/mattermost/mattermost-plugin-msteams-sync/server/msteams"
 	"github.com/mattermost/mattermost-plugin-msteams-sync/server/msteams/clientmodels"
 	"github.com/mattermost/mattermost-plugin-msteams-sync/server/store/storemodels"
-	"github.com/mattermost/mattermost-server/v6/model"
-	"github.com/mattermost/mattermost-server/v6/plugin"
+	"github.com/mattermost/mattermost/server/public/model"
+	"github.com/mattermost/mattermost/server/public/plugin"
 	"github.com/microsoftgraph/msgraph-sdk-go/models"
 	"github.com/pkg/errors"
 	"gitlab.com/golang-commonmark/markdown"
 )
 
+func (p *Plugin) UserWillLogIn(_ *plugin.Context, user *model.User) string {
```

Comment:
I'd like to better understand why we do this /and/ have the sync job do it. If we can promote on login, do we need the sync job to do anything, and could we simplify things accordingly?

Reply:
We need `sync job` to update users authentication methods from `email` to other auth services. I have explained the flow of user login in the [comment](https://github.com/mattermost/mattermost-plugin-msteams-sync/pull/386#discussion_r1400218678).

Reply:
Right, so I better understand now that the job sets all these users to the right authentication type. Can you expand on what it means to clear the `RemoteId` bit? Does that have some effect elsewhere?

Reply:
By clearing the `RemoteId` here, we convert a user from a remote user to a normal user.
> Does that have some effect elsewhere?

There is an edge case here: if another plugin also creates some remote user with some type of authentication method, say SAML, and we have `automaticallyPromoteSyntheticUsers` enabled, the remote users created by other plugins will also be converted to a normal user because of the above hook.

Reply:
cc @wiggin77, as I believe we realized that this directly conflicts with the shared channels functionality. While not currently in use where we use this plugin, do you think this is a blocker or something we can revisit later?

Reply:
@lieut-data we can revisit later, but this makes the plugin incompatible with any installation also using Shared Channels.  With the remoteId cleared, those users will appear to be local and will not sync.